Snippets/Snippets_5/NHibernate/NHibernateLoadMappingsFromFileSystem.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using System.IO;
+using NServiceBus;
+using NServiceBus.Persistence;
+using UnitTesting.SagaTest;
+
+
+class NHibernateCustomMappings
+{
+    #region NHibernateLoadMappingsFromFileSystem
+
+    static void AddMappingsFromFilesystem(NHibernate.Cfg.Configuration nhConfiguration)
+    {
+        var folder = Directory.GetCurrentDirectory();
+        var hmbFiles = Directory.GetFiles(folder, "*.hbm.xml", SearchOption.TopDirectoryOnly);
+
+        foreach (var file in hmbFiles)
+        {
+            nhConfiguration.AddFile(file);
+        }
+    }
+
+    #endregion
+
+    #region NHibernateInitWithFluentNHibernate
+
+    static NHibernate.Cfg.Configuration BuildConfiguration(NHibernate.Cfg.Configuration nhConfiguration)
+    {
+        return FluentNHibernate.Cfg.Fluently.Configure(nhConfiguration)
+            .Mappings(cfg =>
+            {
+                cfg.FluentMappings.AddFromAssemblyOf<MySagaData>();
```

Comment:
That is a good remark. I know that when using `*.hmb.xml` that it is not required to customize all sagas. For the other, don't know. I don't know what NServiceBus does with the NHibernate configuration. Needs some investigation.